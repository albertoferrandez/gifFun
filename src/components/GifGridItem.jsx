import PropTypes from "prop-types"

const GifGridItem = ({id, title, url}) => {
    return (
        <div className="card" key={id}>
            <img src={url} alt={title} />
        </div>
    )
}

export default GifGridItem

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
