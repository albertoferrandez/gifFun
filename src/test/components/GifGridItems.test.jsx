/** @jest-environment jsdom */

import { render, screen } from "@testing-library/react"
import GifGridItem from "../../components/GifGridItem"

describe('pruebas en grifgriditem', () => {

    const title = 'Saitama'
    const url = 'https://one-punch.com/saitama.jpg'

    test('should match con snapshot', () => {
      
        const { container } = render( <GifGridItem title={title} url={url}/> )
        expect(container).toMatchSnapshot()

    })

    test('debe mostrar la imagen con el URL y el ALT indicado', () => {
        render( <GifGridItem title={title} url={url}/> )
        //screen.debug()

        const {src, alt} = screen.getByRole('img')
        expect( src ).toBe(url)
        expect(alt).toBe(alt)
    })

})
