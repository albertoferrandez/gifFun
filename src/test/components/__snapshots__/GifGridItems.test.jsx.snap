// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en grifgriditem should match con snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
  </div>
</div>
`;
